juju/client/connection.py

```diff
@@ -566,6 +586,7 @@ def connect_params(self):
             'bakery_client': self.bakery_client,
             'loop': self.loop,
             'max_frame_size': self.max_frame_size,
+            'proxy': self.proxy,
```

Comment:
Not your mistake, but it looks like the docstring for this `connect_params` function is incorrect. It says it "returns a tuple ... the first element in the tuple" but it actually returns a dict. Maybe you could fix while you're in there?